### PR TITLE
Code Optimization

### PR DESCRIPTION
**Pull request by Gauddel**

### CODE REVIEW

File: contracts/functions/dapps/FCompound.sol [hunk 1]
```diff
@@ -105,6 +105,13 @@ function _getAssetLiquidity(
     return vars;
 }
 
+function _isCompoundUnderlyingLiquidity(address _debtToken, uint256 _debtAmt)
```
Comment (gitpusha) [inline contracts/functions/dapps/FCompound.sol:108]: rename to `_isCompoundUnderlyingLiquid`

File: contracts/functions/dapps/FCompound.sol [hunk 2]
```diff
@@ -105,6 +105,13 @@ function _getAssetLiquidity(
...
+    view
+    returns (bool)
+{
+    return ICToken(_getCToken(_debtToken)).getCash() > _debtAmt;
+}
+
```
Comment (gitpusha) [inline contracts/functions/dapps/FCompound.sol:114]: We already have this function: https://github.com/gelatodigital/gelato-instadapp/blob/master/contracts/functions/gelato/conditions/compound/FCompoundHasLiquidity.sol

File: contracts/functions/gelato/FGelatoDebtBridge.sol [hunk 3]
```diff
@@ -23,23 +23,32 @@ import {
 import {
     IInstaPoolResolver
 } from "../../interfaces/InstaDapp/resolvers/IInstaPoolResolver.sol";
-import {_getMakerVaultDebt} from "../dapps/FMaker.sol";
+import {_getMakerVaultDebt, _debtCeilingIsReached} from "../dapps/FMaker.sol";
+import {_isAaveUnderlyingLiquidV1} from "../dapps/FAave.sol";
+import {_isCompoundUnderlyingLiquidity} from "../dapps/FCompound.sol";
```
Comment (gitpusha) [inline contracts/functions/gelato/FGelatoDebtBridge.sol:28]: We can just import `_cTokenHasLiquidity` from FCompoundHasLiquidity

File: contracts/functions/gelato/conditions/FCanDoRefinance.sol [hunk 4]
```diff
@@ -44,50 +46,62 @@ function _canDoMakerToMakerDebtBridge(DebtBridgeInputData memory _data)
...
+            _data.fees,
+            _data.oracleAggregator
+        );
+
     return
-        _cTokenHasLiquidity(DAI, _data.debtAmt) &&
+        _cTokenHasLiquidity(
+            DAI,
+            _data.flashRoute == 1
+                ? _data.debtAmt + maxBorToCompPos
```
Comment (gitpusha) [inline contracts/functions/gelato/conditions/FCanDoRefinance.sol:98]: why `_data.debtAmt + maxBorToCompPos` if `_data.flashRoue == 1`

https://github.com/gelatodigital/gelato-instadapp/blob/8870f32924ac161a4d5f242d748bd5bc95c794ee/contracts/functions/gelato/FGelatoDebtBridge.sol#L44

**Review by hilmarx — APPROVED** [review]

**Pull request merged**